src/client/ui/panels/AssetExplorerPanelContainer.js

```diff
@@ -188,6 +190,18 @@ class AssetExplorerPanelContainer extends Component {
     document.body.removeChild(textArea);
   };
 
+  onExtend = (e, file) => {
+    if (!file) {
+      return;
+    }
+    if (file.ext === ".gltf") {
+      this.props.editor.signals.openScene.dispatch(file.uri);
+    } else {
+      this.props.editor.extendScene(file.uri);
```

Comment:
You can't call `extendScene` here directly, it has to go through a signal dispatch similar to `openScene` which is handled in EditorContainer so that you can call `confirmSceneChange`, or else you will blow away the user's unsaved changes without notifying them. I would create a new `extendScene` signal for this.